pkg/prometheus/promcfg.go

```diff
@@ -147,6 +147,27 @@ func (cg *ConfigGenerator) AppendMapItem(m yaml.MapSlice, k string, v interface{
 	return append(m, yaml.MapItem{Key: k, Value: v})
 }
 
+// AppendCommandlineArgument appends the name/v argument to the given []monitoringv1.Argument and returns
+// the updated slice.
+func (cg *ConfigGenerator) AppendCommandlineArgument(m []monitoringv1.Argument, argument monitoringv1.Argument) []monitoringv1.Argument {
+	if cg.notCompatible {
+		level.Warn(cg.logger).Log("msg", fmt.Sprintf("ignoring command line argument %q not supported by Prometheus", argument.Name))
+		return m
+	}
+
+	return append(m, argument)
+}
+
+// IsCompatible return true or false depending if the version being used is compatible
+func (cg *ConfigGenerator) IsCompatible(name string) bool {
```

Comment:
At the moment this only avoids one `if/else` on line 539 of promehteus/stefulset.go, I'm also up for removing it and leaving the `if/else`

Reply:
sounds good to me.
Maybe modify the function to make it more obvious that it can also log a warning?

```suggestion
func (cg *ConfigGenerator) IsCompatible(field string) bool {
    return cg.notCompatible
}

func (cg *ConfigGenerator) IsCompatibleOrWarn(field string) bool {
...
}
```